JavaScript/chapter01/p05_oneaway/avc278.js

```diff
@@ -0,0 +1,80 @@
+// One Away: There are three types of edits that can be performed on strings: insert a character, remove a character,
+// or replace a character. Given two strings, write a function to check if they are one edit (or zero edits) away.
+
+const assert = require("assert");
+
+/**
+ * Check if the two strings are zero or one edits away
+ * @param  {string}  A input string one to check against
+ * @param  {string}  B input string two to check against
+ * @return {boolean}   whether the input strings can be edited zero or one ways to be equal
+ *
+ * There are two cases here: (1) A.length === B.length, and (2) A.length !== B.length.
+ * For the first case, all we need to do is iterate through the strings and ensure their elements are equal, allowing
+ * room for one misplaced character. Iterating through this takes O(N) time, without any additional space.
+ * For the second case, we iterate through the different length strings, and perform almost identical comparisons.
+ * Again, we allow a single misplaced character, or lack of character as we iterate through them. This also takes O(N)
+ * time. For the sake of keeping things DRY, we find out which string is the shorter one and instantiate a copy of it,
+ * as well as a copy of the long string. For this reason, we take additional O(N) space. If we wanted, we could perform
+ * two while loops (a < A.length + 1 && b < B.length) as well as (a < A.length && b < B.length + 1), so we don't take
+ * any additional space.
+ * Runtime: O(N)
+ * Space:   O(N)
+ *
+ */
+const oneAway = (A, B) => {
+  if (Math.abs(A.length - B.lenght) > 1) return false;
```

Comment:
```suggestion
  if (Math.abs(A.length - B.length) > 1) return false;
```

huh, I wonder how this is passing the tests.

Reply:
Well that's embarrassing.
Let's pretend no one saw that :)